sympy/physics/mechanics/body.py

```diff
@@ -0,0 +1,170 @@
+        self.loads = []
+
+        if frame is None:
+            frame = ReferenceFrame(name + '_frame')
+
+        if masscenter is None:
+            masscenter = Point(name + '_masscenter')
+
+        if central_inertia is None and mass is None:
+            ixx = Symbol(name + '_ixx')
+            iyy = Symbol(name + '_iyy')
+            izz = Symbol(name + '_izz')
+            izx = Symbol(name + '_izx')
+            ixy = Symbol(name + '_ixy')
+            iyz = Symbol(name + '_iyz')
+            _inertia = (inertia(frame, ixx, iyy, izz, ixy, iyz, izx),
+                        masscenter)
+        else:
+            _inertia = (central_inertia, masscenter)
+
+        if mass is None:
+            _mass = Symbol(name + '_mass')
+        else:
+            _mass = mass
+
+        masscenter.set_vel(frame, 0)
+
+        # If user passes masscenter and mass then a particle is created
+        # otherwise a rigidbody. As a result a body may or may not have inertia.
+        if central_inertia is None and mass is not None:
+            self.frame = frame
+            self.masscenter = masscenter
+            Particle.__init__(self, name, masscenter, _mass)
+        else:
+            RigidBody.__init__(self, name, masscenter, frame, _mass, _inertia)
+
+    def apply_force(self, vec, point=None):
+        """
+        Adds the force to the point (center of mass by default) on the body.
+
+        Parameters
+        ----------
+        vec: Vector
+            Defines the force vector. Can be any vector w.r.t any frame or
+            combinations of frame.
+        point: Point, optional
+            Defines the point on which the force must be applied. Default is
+            Body's center of mass.
+
+        Example
+        -------
+        To apply a unit force in x direction of body's frame to body's
+        center of mass.
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Body
+        >>> body = Body('body')
+        >>> g = Symbol('g')
+        >>> body.apply_force(body.mass * g * body.frame.x)
+
+        To apply force to any other point than center of mass, pass that point
+        as well.
+
+        >>> from sympy import Symbol
+        >>> from sympy.physics.mechanics import Body
+        >>> parent = Body('parent')
+        >>> child = Body('child')
+        >>> g = Symbol('g')
+        >>> frame = parent.frame
+        >>> l = Symbol('l')
+        >>> point = child.masscenter.locatenew('force_point', l * body.frame.y)
+        >>> gravity = child.mass * g
+        >>> body.apply_force(gravity * body.frame.x, point)
+
+        """
+        if not isinstance(point, Point):
+            if point is None:
+                point = self.masscenter  # masscenter
+            else:
+                raise TypeError("A Point must be supplied to apply force to.")
+        if not isinstance(vec, Vector):
+            raise TypeError("A Vector must be supplied to apply force.")
```

Comment:
Lines 148-154 are not unit tested. (Run `bin/coverage_report.py sympy/physics/mechanics`) to see.